src/root/usr/bin/run-postgresql

```diff
@@ -20,8 +20,14 @@ generate_postgresql_config
 # Is this brand new data volume?
 PG_INITIALIZED=false
 
-if [ ! -f "$PGDATA/postgresql.conf" ]; then
-  initialize_database
+if [ ! -f "${PGDATA}/postgresql.conf" ]; then
+  # Minimize the race condition of unfinished DB initialization vs pod restart
+  # First initialize the DB into temporary directory
+  TEMP_PGDATA=$(mktemp -d $(printf '%0.sX' {1..32}))
```

Comment:
Preferrably `TEMP_PGDATA=$(mktemp -d "$(printf '%0.sX' {1..32})")`.

Reply:
Btw., do we have to have the printf magic at all?

Reply:
@praiskup That printf magic is there just to have 32-times the _X_ character (the final temporary dir to be "strong enough", not guessable). But can replace it with the final form (since better to understand later).